catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/SolrMetacardClientImpl.java

```diff
@@ -70,6 +70,18 @@
 
     private static final String DISTANCE_SORT_FIELD = "_distance_";
 
+    private static final String SOLR_QUERY_ROWS_PROPERTY = "solr.query.rows";
+
+    /**
+     * Because {@link SolrQuery#setRows(Integer)} will increment the given integer and use it as an
+     * array index in Solr Cloud mode, we cannot use {@link Integer#MAX_VALUE} by default and should
+     * pick a much lower, sensible ceiling for our row count. In practice, an un-paged query with a
```

Comment:
Will remove: `In practice, an un-paged query ...` since that is subjective to the use case. 
Code has been changed to provide 2 million entries worth of space which can be lowered if necessary under rare circumstances (i.e. an implementation uses unpaged queries and can't be changed).